### PR TITLE
feat: add error message for duplicate flags in `no-invalid-regexp`

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
Added a separate error message for duplicate flags (only those are valid and allowed flags) in `no-invalid-regexp` rule.

### CODE REVIEW

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

Leaving open since this is marked as a feature so it requires additional approval from any other team member. Also, since this PR was opened on Saturday, it needs to wait until Tuesday before merging.

File: lib/rules/no-invalid-regexp.js
Comment (nzakas) [inline lib/rules/no-invalid-regexp.js:153]: Could we reformat this message? It reads a bit strange to me.
```suggestion
                return `Duplicate flags ('${duplicateFlags.join("")}') supplied to RegExp constructor.`;
```

**Review by nzakas — APPROVED** [review]
LGTM. Just leaving open until Tuesday per our merging rules.

**Review by mdjermanovic — APPROVED** [review]

**Pull request merged**
